app/src/main/java/com/github/sdp/tarjetakuna/utils/ExportCollection.kt

```diff
@@ -0,0 +1,121 @@
+package com.github.sdp.tarjetakuna.utils
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Environment
+import android.view.View
+import androidx.core.content.ContextCompat.startActivity
+import androidx.core.content.FileProvider
+import com.github.sdp.tarjetakuna.R
+import com.github.sdp.tarjetakuna.model.MagicCard
+import com.google.android.material.snackbar.Snackbar
+import org.apache.poi.hssf.usermodel.HSSFWorkbook
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.ss.usermodel.Sheet
+import org.apache.poi.ss.usermodel.Workbook
+import org.apache.poi.ss.util.CellUtil
+import java.io.File
+import java.io.FileOutputStream
+
+/**
+ * This class is used to export the user's collection to an excel file
+ */
+object ExportCollection {
+
+    private const val fileProviderAuthority = "com.github.sdp.tarjetakuna.fileprovider"
+    private const val fileName = "MyMagicCollection.xls"
+    private const val sheetName = "MyMagicCollection"
+    private const val applicationType = "application/excel"
+
+    /**
+     * Export the user's collection to an excel file
+     */
+    fun exportCollection(view: View, collectionToExport: List<MagicCard>) {
+        val filePath: String
+        try {
+            filePath =
+                view.context.getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS)!!.absolutePath + "/" + fileName
+        } catch (e: Exception) {
+            Snackbar.make(
+                view,
+                R.string.ExportCollection_fileDirectoryNotFound,
+                Snackbar.LENGTH_LONG
+            ).show()
+            return
+        }
+
+        val excelFile = File(filePath)
+        if (writeDataToExcel(view, excelFile, collectionToExport)) openExcelFile(

```

Comment:
need a toast or something to say that the file as been created, and that you might want to share it. 
As it is now, it feels like it's never exported as a local file at all until you share it

Reply:
And it is. See the comment I made above.

Reply:
aggreed, to bad it s not a permanent local file, but it kinda make sense. I would be happy to not have to share it directly, but just export it locally as well as it was before